data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/ownership/ConfigBucketOwnerProviderFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.ownership;
+
+import com.amazon.dataprepper.plugins.source.S3SourceConfig;
+import com.amazon.dataprepper.plugins.source.SqsQueueUrl;
+
+import java.net.MalformedURLException;
+
+/**
+ * Produces a {@link BucketOwnerProvider} from the S3 source configuration as
+ * provided in a {@link S3SourceConfig}.
+ */
+public class ConfigBucketOwnerProviderFactory {
+    /**
+     * Creates the {@link BucketOwnerProvider}
+     * @param s3SourceConfig The input {@link S3SourceConfig}
+     * @return The {@link BucketOwnerProvider}
+     */
+    public BucketOwnerProvider createBucketOwnerProvider(final S3SourceConfig s3SourceConfig) {
+        if(s3SourceConfig.isDisableBucketOwnershipValidation())
+            return new NoOwnershipBucketOwnerProvider();
```

Comment:
nit: I see BucketOwnerProvider is annotated as FunctionalInterface. We could return lambda function instead of defining separate class unless we plan to reference or reuse NoOwnershipBucketOwnerProvider in other parts of the code.